flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -179,6 +187,22 @@ public static String getCommonStartCommand(
 		return BootstrapTools.getStartCommand(commandTemplate, startCommandValues);
 	}
 
+	public static File checkJarFileForApplicationMode(Configuration configuration) {
```

Comment:
Nice suggestion. I will create a new class `ApplicationUtils` and move `resolveURI` and `checkLocalJarFilesForApplicationMode` in since they could be used by other deployment(e.g. Yarn application).